File: docs/src/testers/functions/invocations/async_create_start.py
# @sniptest filename=async_create_start.py
import os

from notte_sdk import NotteClient

client = NotteClient()
function = client.Function(os.environ["NOTTE_FUNCTION_ID"])
# Optional: create a run record when you need its ID before execution.
created = function.create_run()

run_id = created.function_run_id
print(f"Run created: {run_id}")

# Execute that run and wait for its result.
result = function.run(function_run_id=run_id, url="https://example.com")
print(result.result)
run = function.get_run(run_id)
print(run.status)

File: docs/src/testers/functions/invocations/async_create_start.ts
// @sniptest filename=async_create_start.ts
import { NotteClient } from 'notte-sdk';

const client = new NotteClient();
const fn = client.NotteFunction({ function_id: process.env.NOTTE_FUNCTION_ID! });

// Optional: create a run record when you need its ID before execution.
const created = await fn.createRun();
console.log(`Run created: ${created.function_run_id}`);

// Execute that run and wait for its result. Omit functionRunId to create a new run.
const result = await fn.run(
  { url: 'https://example.com' },
  { functionRunId: created.function_run_id },
);
console.log(result.result);
const run = await fn.getRun(created.function_run_id);
console.log(run.status);
